lib/openhab/core/item_proxy.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'delegate'
+require 'forwardable'
+
+module OpenHAB
+  module Core
+    # Class is a proxy to underlying item
+    class ItemProxy < Delegator
+      extend Forwardable
+      def_delegators :__getobj__, :class, :is_a?, :tap, :then, :yield_self, :hash, :kind_of, :to_s
```

Comment:
Why are you forwarding tap, then, and yield_self to the actual item? If it's because BasicObject (and thus SimpleDelegator) don't have those methods, you should re-implement them so that the actual item doesn't escape.

Reply:
tap/then/and yield_self were probably not correct to forward as they should be handled by the delegator.  I *think* we need to forward most methods in 'Module' i.e. class, is_a?, instance_of?, etc.  but I was mostly looking at the list of methods in Module and guessing.

Reply:
yeah the rest of the list looked correct to me